src/dreamchecker/lib.rs

```diff
@@ -69,6 +81,19 @@ impl<'o> StaticType<'o> {
     fn list_of_type(tree: &'o ObjectTree, of: &str) -> StaticType<'o> {
         StaticType::List { list: tree.expect("/list"), keys: Box::new(StaticType::Type(tree.expect(of))) }
     }
+
+    fn is_subtype_of(&self, other: &StaticType<'o>) -> bool {
+        match (self, other) {
+            (StaticType::Type(selftype), StaticType::Type(othertype)) => {
+                selftype.is_subtype_of(othertype.get())
+            },
+            (StaticType::List{list: selflist, keys: _}, StaticType::List{list: otherlist, keys: _}) => {
+                selflist.is_subtype_of(otherlist.get())
```

Comment:
you probably want to be comparing `keys`, not `list`. `list` is always initialized to a reference to `/list`

Reply:
suddenly this enum makes way more sense